samples: nrf5340: extxip_smp_svr: Convert to sysbuild (#16513)
nrfconnect/sdk-nrf: doc/nrf/config_and_build/bootloaders/qspi_xip_split_image.rst

.. _qspi_xip_split_image:

Using QSPI XIP split image
##########################

.. contents::
   :local:
   :depth: 2

The QSPI XIP split image feature lets you gain more flash storage space for applications by splitting application code into two parts:

> * Code that runs on the internal flash.

Reviewer: ```suggestion
* Code that runs on the internal flash memory.
```

Author: Fixed

> * Code that runs on a supported external flash over Quad Serial Peripheral Interface (QSPI) using Execute in place (XIP).

Reviewer: ```suggestion
* Code that runs on supported external flash memory over the Quad Serial Peripheral Interface (QSPI) using Execute in Place (XIP).
```

Reviewer 2: I was leaning towards your edit, but then I saw it is spelled in sentence style in the PS https://docs.nordicsemi.com/bundle/ps_nrf52840/page/qspi.html#ariaid-title6

Author: Fixed

This feature is supported on nRF52840, nRF5340, and the SoCs of the nRF91 Series.

.. warning::
    On the nRF52840, do not relocate interrupts to the QSPI XIP flash.
    Doing so can lock up or brick the device by making the debug access port inaccessible.

The QSPI XIP split images are supported in :doc:`MCUboot <mcuboot:index-ncs>`, which allows for updating them over-the-air.

For the nRF5340 DK and Nordic Thingy:53, you can also check out the :ref:`SMP Server with external XIP <smp_svr_ext_xip>` sample, which demonstrates this feature.

Requirements
************

Make sure you meet the following requirements to make use of this feature:

> * Board based on nRF52840, nRF5340, or nRF91 Series SoCs, compatible with :ref:`Hardware model v2 (HWMv2) <hwmv1_to_v2_migration>`

Reviewer: Is the hardware board or the related board files that have to be compatible with hwmodelv2?

Can a physical board not be compatible with hwmodelv2? Isn't this just a software configuration issue? (meaning, edited board files are needed)

Reviewer 2: ```suggestion
* Board based on nRF52840, nRF5340, or nRF91 Series SoCs, with file structure compatible with :ref:`Hardware model v2 (HWMv2) <hwmv1_to_v2_migration>`
```

Author: hwmv1 is just the board files themselves, not overlay files
